index.js

```diff
@@ -111,6 +116,44 @@ Babel.prototype.optionsHash = function() {
       hash[key] = (typeof value === 'function') ? (value + '') : value;
     }
 
+    if (options.plugins) {
+      hash.plugins = [];
+
+      for (var i = 0; i < options.plugins.length; i++) {
+        var plugin = options.plugins[i];
+
+        var pluginType = typeof plugin;
+        var pluginAugmentsCacheKey = false;
+        var pluginProvidesBaseDir = false;
+
+        // do not look for properties on string plugins
+        if (pluginType === 'function' || pluginType === 'object') {
+          pluginAugmentsCacheKey = typeof plugin.cacheKey === 'function';
+          pluginProvidesBaseDir = typeof plugin.baseDir === 'function';
+        }
+
+        if (pluginAugmentsCacheKey) {
+          hash.plugins.push(plugin.cacheKey());
+        }
+
+        if (pluginProvidesBaseDir) {
+          var depHash = hashForDep(plugin.baseDir());
+
+          hash.plugins.push(depHash);
+        }
+
+        if (pluginType === 'string') {
+          hash.plugins.push(plugin);
+        } else if (!pluginProvidesBaseDir){
+          // prevent caching completely if the plugin doesn't provide baseDir
+          // we cannot ensure that we aren't causing invalid caching pain...
+          this.console.log('broccoli-babel-transpiler is opting out of caching due to a plugin that does not provide a caching strategy: `' + plugin + '`.');
+          hash.plugins.push((new Date).getTime() + Math.random());
```

Comment:
There is no else, so either it's a string plugin (and takes no options) or it has a baseDir. Anything else gets the non-persistent treatment.  

Also, RE: the issue in b-p-f I'm happy to work on that next. I agree switching back out of persistent strat is better than what I'm doing here.